sdk/core/azure-core/inc/nullable.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <new> // for placement new
+#include <stdlib.h> // for abort
```

Comment:
```suggestion
#include <cstdlib> // for abort
```

Reply:
> > > Wouldn't be better to have the implementation similar to `std::optional` and if C++ 17 is available, default to `std::optional`?
> > > https://github.com/microsoft/STL/blob/master/stl/inc/optional
> > 
> > 
> > We cannot use C++17. The latest C++ standard we can use is C++14
> 
> [Bond](https://github.com/microsoft/bond) needs to support C++ 11 and up. There, I created a conditional wrapper to support C++ 17 conditional code:
> https://github.com/bhardwajs/bond/blob/e83de6ffe5fcb38fb3e9f637351f3f9b90479dde/cpp/inc/bond/core/config.h#L35
> 
> Similarly, this can be
> 
> ```
> #if CXX_17
> using Nullable<T> = std::optional<T>;
> #else
> ...
> #endif
> ```

That is correct that we can't have C++17 dependencies.  Nullable<T> is slimmer version of optional<T> and not 100% compatible with std::optional<T>.  The above MACRO would imply either works in all situations and that is not true.